src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,190 @@
+                        final boolean logRequestBody,
+                        final boolean resolveIndices,
+                        final boolean excludeSensitiveHeaders,
+                        final Set<String> ignoredAuditUsers,
+                        final Set<String> ignoredComplianceUsersForRead,
+                        final Set<String> ignoredComplianceUsersForWrite,
+                        final Set<String> ignoredAuditRequests,
+                        final EnumSet<AuditCategory> disabledRestCategories,
+                        final EnumSet<AuditCategory> disabledTransportCategories,
+                        final String opendistrosecurityIndex) {
+        this.isRestApiAuditEnabled = isRestApiAuditEnabled;
+        this.isTransportAuditEnabled = isTransportAuditEnabled;
+        this.resolveBulkRequests = resolveBulkRequests;
+        this.logRequestBody = logRequestBody;
+        this.resolveIndices = resolveIndices;
+        this.excludeSensitiveHeaders = excludeSensitiveHeaders;
+        this.ignoredAuditUsers = ignoredAuditUsers;
+        this.ignoredComplianceUsersForRead = ignoredComplianceUsersForRead;
+        this.ignoredComplianceUsersForWrite = ignoredComplianceUsersForWrite;
+        this.ignoreAuditRequests = ignoredAuditRequests;
+        this.disabledRestCategories = disabledRestCategories;
+        this.disabledTransportCategories = disabledTransportCategories;
+        this.opendistrosecurityIndex = opendistrosecurityIndex;
+    }
+
+    public static AuditConfig getConfig(Settings settings) {
+        final boolean isRestApiAuditEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_REST, true);
+        final boolean isTransportAuditEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_ENABLE_TRANSPORT, true);
+        final boolean resolveBulkRequests = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_BULK_REQUESTS, false);
+        final boolean logRequestBody = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, true);
+        final boolean resolveIndices = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true);
+        final boolean excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+        final String opendistrosecurityIndex = settings.get(ConfigConstants.OPENDISTRO_SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        final EnumSet<AuditCategory> disabledRestCategories;
+        final List<String> disabledRestCategoriesList = settings.getAsList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, DEFAULT_DISABLED_CATEGORIES);
+        if (disabledRestCategoriesList.isEmpty() || (disabledRestCategoriesList.size() == 1 && "NONE".equalsIgnoreCase(disabledRestCategoriesList.get(0)))) {
```

Comment:
I would appreciate if you could help design this helper function. Previously I had 

```
private static List<String> getSettingAsList(final Settings settings, final String key, final List<String> defaultList) {
        final List<String> list = settings.getAsList(key, defaultList);
        if (list.size() == 1 && "NONE".equals(list.get(0))) {
            return Collections.emptyList();
        }
        return list;
    }
```

Looks like other params consider only "NONE" as implicitly empty (none, None are not considered). Do you want to retain this behavior or clear it in #325

Reply:
One way is to pass a flag whether "NONE" is case sensitive or not. Assuming that "NONE" will be deprecated soon, I suggest retaining the current behavior.

Reply:
Alright!
Sadly, I feel case in-sensitive handling of none was accidental in categories because of making it uppercase for enums and now have to deal with complexity.